python_src/src/lamp_py/performance_manager/gtfs_utils.py

```diff
@@ -79,58 +78,67 @@ def add_fk_static_timestamp_column(
     process_logger = ProcessLogger(
         "add_fk_static_timestamp",
         row_count=events_dataframe.shape[0],
-        timestamp_column=timestamp_column,
     )
     process_logger.log_start()
 
-    # get unique "start_date" values from events dataframe
-    # with associated minimum value of timestamp column
-    date_groups = events_dataframe.groupby(by="start_date")[
-        timestamp_column
-    ].min()
+    # initialize fk_static_timestamp column
+    events_dataframe["fk_static_timestamp"] = 0
 
-    # dictionary used to match minimum timestamp column values to
-    # "timestamp" from StaticFeedInfo table, to be used as foreign key
-    timestamp_lookup = {}
-    for date, min_timestamp in date_groups.items():
+    for date in events_dataframe["start_date"].unique():
         date = int(date)
-        min_timestamp = int(min_timestamp)
         # "start_date" from events dataframe must be between "feed_start_date" and "feed_end_date" in StaticFeedInfo
-        # minimum timestamp column must also be less than "timestamp" from
-        # StaticFeedInfo, order by "timestamp" descending and limit to 1 result
-        # this should deal with multiple static schedules with possible
-        # overlapping times of applicability
-        feed_timestamp_query = (
+        # "start_date" must also be less than or equal to "feed_active_date" in StaticFeedInfo
+        # StaticFeedInfo, order by feed_active_date descending and created_on date descending
+        # this should deal with multiple static schedules being issued on the same day
+        # if this occurs we will use the latest issued schedule
+        live_match_query = (
             sa.select(StaticFeedInfo.timestamp)
             .where(
-                (StaticFeedInfo.feed_start_date <= date)
-                & (StaticFeedInfo.feed_end_date >= date)
-                # & (StaticFeedInfo.timestamp < min_timestamp)
+                StaticFeedInfo.feed_start_date <= date,
+                StaticFeedInfo.feed_end_date >= date,
+                StaticFeedInfo.feed_active_date <= date,
+            )
+            .order_by(
+                StaticFeedInfo.feed_active_date.desc(),
+                StaticFeedInfo.created_on.desc(),
+            )
+            .limit(1)
+        )
```

Comment:
This initial query will match to GTFS-static schedules we are processing in real time with the `feed_active_date` field from FEED_INFO